src/states/factory.ts

```diff
@@ -0,0 +1,40 @@
+// need updated ts states file from other task
+import * as states from './states';
+
+/**
+ * creates an instance of a specified class with default values
+ *
+ * @param {string} dataType - the dataType from the object returned by getDataType.ts
+ * @param {string} name - the name from the object returned by getDataType.ts
+ * @returns {BaseState} - instance of the state class from the data type with corresponding name
+ */
```

Comment:
Since we are now doing TS we do not need the type defined in JSDoc comments.
```suggestion
/**
 * creates an instance of a specified class with default values
 *
 * @param dataType The dataType from the object returned by getDataType.ts
 * @param name The name from the object returned by getDataType.ts
 * @returns Instance of the state class from the data type with corresponding name
 */
```